### ISSUE TITLE
go.discovery.com HTTP Error 403: Forbidden

### ISSUE THREAD

**Issue by JBGuy44** [yt-dlp/yt-dlp#5507]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

URL= https://go.discovery.com/video/urban-legend-travel-channel-atve-us/the-red-room

I am attempting to download a video from go.discovery.com.

I am able to play the URL in both Chrome and Firefox.

I am using --cookies-from-browser.

Returning HTTPERROR 403: 'Forbidden'

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
E:\yt-dlp>yt-dlp -vU --cookies-from-browser Chrome https://go.discovery.com/video/urban-legend-travel-channel-atve-us/the-red-room
[debug] Command-line config: ['-vU', '--cookies-from-browser', 'Chrome', 'https://go.discovery.com/video/urban-legend-travel-channel-atve-us/the-red-room']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-100139-g456146356a, ffprobe N-100139-g456146356a
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[Cookies] Extracting cookies from chrome
[debug] Extracting cookies from: "C:\Users\Jack\AppData\Local\Google\Chrome\User Data\Default\Network\Cookies"
[debug] Found local state file at "C:\Users\Jack\AppData\Local\Google\Chrome\User Data\Local State"
[Cookies] Extracted 7257 cookies from chrome
[debug] cookie version breakdown: {'v10': 7257, 'other': 0, 'unencrypted': 0}
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [GoDiscovery] Extracting URL: https://go.discovery.com/video/urban-legend-travel-channel-atve-us/the-red-room
[debug] Using fake IP 6.15.119.127 (US) as X-Forwarded-For
[GoDiscovery] urban-legend-travel-channel-atve-us/the-red-room: Downloading JSON metadata
ERROR: [GoDiscovery] urban-legend-travel-channel-atve-us/the-red-room: Unable to download JSON metadata: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 674, in extract
  File "yt_dlp\extractor\dplay.py", line 367, in _real_extract
  File "yt_dlp\extractor\dplay.py", line 78, in _get_disco_api_info
  File "yt_dlp\extractor\common.py", line 1034, in download_content
  File "yt_dlp\extractor\common.py", line 998, in download_handle
  File "yt_dlp\extractor\common.py", line 868, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 825, in _request_webpage

  File "yt_dlp\extractor\common.py", line 807, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 3692, in urlopen
  File "urllib\request.py", line 531, in open
  File "urllib\request.py", line 640, in http_response
  File "urllib\request.py", line 569, in error
  File "urllib\request.py", line 502, in _call_chain
  File "urllib\request.py", line 649, in http_error_default
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by rredford6** [comment]
@JBGuy44 `--print-traffic` would help a lot here.

----

This does not seem to be an issue with --cookies or authenticated videos.

One of the sample URLs from this extractor:
https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L373
is currently free.  Opening it in a browser _without_ linking a provider works fine.

Tracing the code, here's what I see:
1. [`_real_extract`](https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L312) is the entry point.
2. [`_get_disco_api_info`](https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L67) is called to retrieve some basic information about the video.  This is the `Downloading JSON metadata` step.
3. Some headers are set.  Irrelevant for what causes the actual error.
4. [`_update_disco_api_headers`](https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L52) is called.  This is where the misbehaving starts.
5. `headers['Authorization']` is set to the output of `_get_auth`.  The API *requires* an authorization header to download the JSON metadata.  All calls will fail without it.  This includes videos that are free to watch.
6. [`_get_auth`](https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L22) gets the authorization token.  In your case, this will parse cookies.  In the case where cookies aren't being used, this will hit the token endpoint to generate a fresh token that can be used to download the JSON metadata.
7. Regardless of what token was generated, the `Authorization` header does *not* get sent to the API.  Other declared headers _are_ sent ([here](https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L80)).

I believe this is a bug on line 76:
https://github.com/yt-dlp/yt-dlp/blob/0a4b2f4180b57f8e82b5d9c078c070ddfac7c727/yt_dlp/extractor/dplay.py#L76
The Authorization header generated by `_update_disco_api_headers` is never assigned back to the `headers` object that is sent with the API request.  This code was introduced in 5118d2ec58e3a882900fef7a0191c465ddf3d592 and would likely have been broken for the last 12 months or whenever Scripps started requiring authentication on the info endpoint.  I can definitely recall a time when auth was not required on this endpoint.

With `youtube-dlp --verbose --print-traffic --no-geo-bypass https://go.discovery.com/video/dirty-jobs-discovery-atve-us/rodbuster-galvanizer` I see the following output:
```plain
[debug] [GoDiscovery] Extracting URL: https://go.discovery.com/video/dirty-jobs-discovery-atve-us/rodbuster-galvanizer
[GoDiscovery] dirty-jobs-discovery-atve-us/rodbuster-galvanizer: Downloading JSON metadata
send: b'GET /content/videos/dirty-jobs-discovery-atve-us/rodbuster-galvanizer?fields%5Bchannel%5D=name&fields%5Bimage%5D=height%2Csrc%2Cwidth&fields%5Bshow%5D=name&fields%5Btag%5D=name&fields%5Bvideo%5D=description%2CepisodeNumber%2Cname%2CpublishStart%2CseasonNumber%2CvideoDuration&include=images%2CprimaryChannel%2Cshow%2Ctags HTTP/1.1\r\nHost: us1-prod-direct.go.discovery.com\r\nReferer: https://go.discovery.com/video/dirty-jobs-discovery-atve-us/rodbuster-galvanizer\r\nX-Disco-Client: WEB:UNKNOWN:dsc:25.2.6\r\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nAccept-Language: en-us,en;q=0.5\r\nSec-Fetch-Mode: navigate\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: close\r\n\r\n'
reply: 'HTTP/1.1 400 Bad Request\r\n'
````
Note there is no `Authorization` header that `_update_disco_api_headers` generated..  Scripp's API will accept either an Authorization header or Cookies.  So my recommendation would be to pass cookies through verbatim when they are specified, and only use Authorization when there is no cookie store being read.  Calls in the browser always use Cookies, regardless of whether a TV provider has been linked or not.

@Ashish0804 or @pukkandan can you confirm my suspicion or tell me I need to GTFO and install pdb?

**Comment by rredford6** [comment]
It looks like @timendum tried to fix this with a030af2306cadc3225f83b1ec7217d001f644db8 but #4905 was never merged

**Comment by pukkandan** [comment]
The URL does not open for me, likely due to geo-restriction. I'm unable to check with a VPN atm. But your analysis does look right. Whether that is the issue, I cannot say. Does the linked patch work?

Assuming it does, could u make a new PR? Note that https://github.com/yt-dlp/yt-dlp/pull/4905#issuecomment-1258319964 still applies. If you are unable to test for the other subclasses, I'd instead suggest to make changes only to the specific extractors you can test on (or has an open issue) - we don't want any currently working IE to break

PS: Ashish is no longer active and is unlikely to respond

**Comment by Mr-Jake** [comment]
It works for me. I don't get any 403 error with the sample link in the original post.

```
[GoDiscovery] urban-legend-travel-channel-atve-us/the-red-room: Downloading JSON metadata
[GoDiscovery] 4989104: Downloading JSON metadata
[GoDiscovery] urban-legend-travel-channel-atve-us/the-red-room: Downloading m3u8 information
[info] Available formats for 4989104:
ID       EXT RESOLUTION FPS |   FILESIZE   TBR PROTO | VCODEC        VBR ACODEC
   ABR
--------------------------------------------------------------------------------------
hls-255  mp4 256x144     15 | ~ 78.67MiB  256k m3u8  | avc1.4d000d  256k mp4a.40.5  0k
hls-468  mp4 416x234     30 | ~144.23MiB  469k m3u8  | avc1.4d000d  469k mp4a.40.5  0k
hls-955  mp4 640x360     30 | ~293.80MiB  955k m3u8  | avc1.4d001e  955k mp4a.40.5  0k
hls-2140 mp4 960x540     30 | ~658.49MiB 2141k m3u8  | avc1.4d001f 2141k mp4a.40.5  0k
hls-3375 mp4 1280x720    30 | ~  1.01GiB 3375k m3u8  | avc1.64001f 3375k mp4a.40.5  0k
hls-5236 mp4 1920x1080   30 | ~  1.57GiB 5236k m3u8  | avc1.640028 5236k mp4a.40.5  0k
```

**Comment by rredford6** [comment]
@Mr-Jake Patched or unpatched works for you?

**Comment by Mr-Jake** [comment]
Unpatched. I am using yt-dlp 2022.10.04. The compiled windows exe from the releases section. I have not upgraded to 2022.11.11 yet.

I use the firefox extension cookies.txt to generate my cookie file.

**Comment by bashonly** [comment]
Fixed in 39e6c4cb44b9292e89ac0afec3cd0afc2ae8775f
